nixos/modules/security/sudo.nix

```diff
@@ -207,10 +209,12 @@ in
       ''
         # Don't edit this file. Set the NixOS options ‘security.sudo.configFile’
         # or ‘security.sudo.extraRules’ instead.
-
+      ''
+      (optionalString enableSSHAgentAuth ''
```

Comment:
IMO that's the wrong way around - if we make this optional, then modules/security/pam.nix should insert this config via security.sudo.extraConfig, instead of making the sudo module change its behavior based on options defined in other modules.